unittests/InstructionCountCI/AFP/Secondary_REP.json

```diff
@@ -139,30 +139,33 @@
       ]
     },
     "cmpss xmm0, xmm1, 0": {
-      "ExpectedInstructionCount": 1,
+      "ExpectedInstructionCount": 2,
       "Comment": [
         "0xf3 0x0f 0xc2"
       ],
       "ExpectedArm64ASM": [
-        "fcmeq s16, s16, s17"
+        "fcmeq s0, s16, s17",
```

Comment:
I'm confused with the changes to fcmeq, fcmgt, fcmge here. According to the ARM manual's pseudocode, the scalar version of these instructions should properly merge when NEP is enabled.
```
boolean merge = elements == 1 && IsMerging(FPCR); ///< IsMerging checking FPCR.NEP
bits(128) result = if merge then V[m, 128] else Zeros(128);
```

It just uses vm as the merge register, so we need to be careful.

- cmpss - EQ - swap the arguments
- cmpss - LT - swap arguments and convert to fcmge?
- cmpss - LE - swap arguments and convert to fcmgt?

Reply:
You're right - we can do that. I was so focused into the details of where the top bits were coming from that it didn't occur to me that swapping the arguments would allow us to continue generating the same code we had before. 🤦🏼 

Thanks for the feedback on the patch.

Reply:
@Sonicadvance1 For cmpss, swapping seems to work but I get problems with cmpsd as expected. I will see if there's anything that can be done in that case.

Reply:
Curious, the pseudocode for fcmeq doesn't claim there is a behaviour change between single and double precision scalar.

Reply:
The comment for `IsMerging` in the spec sounds like it's in contradiction to what the NEP bit part says. The IsMerging says:
```
// IsMerging()
// ===========
// Returns TRUE if the output elements other than the lowest are taken from
// the destination register.
```

but the NEP.bit part actually has a bunch of expections where the output elements other than the lowest bit are taken from different places.

![Screenshot from 2024-06-12 16-45-09](https://github.com/FEX-Emu/FEX/assets/7911/a5194b7d-60de-437e-9064-e0d18ce345ad)

Reply:
And the pseudocode for the instruction which shows expected behaviour.
`elements = 1` for the scalar variants.
![Screenshot 2024-06-17 at 02 12 58](https://github.com/FEX-Emu/FEX/assets/1018829/b8342311-e817-4e5d-b11c-64980b5094a0)

Reply:
That's odd - but there are couple of other odd things that I still need to wrap my head around. For example, 
1. I can't see a different in execution path between cmpss and cmpsd. 
2. We are already swapping argumnets by converting cmp-LT into fcmgt and cmp-LE into fcmge. In my head, this cannot work. `LT a1, a2 ==> GE a2, a1` and `LE a1, a2 ==> GT a2, a1`. I am probably missing something but need to understand what.

Reply:
Actually, this transformation might be because of this confusing table.
![Image_2024-06-17_03-09-02](https://github.com/FEX-Emu/FEX/assets/1018829/fd259fcb-1a25-4181-81a4-b067d74b76be)
So that sounds like you could have found a legitimate bug in our implementation.

Reply:
OK - I initially had some misunderstandings about the code but I got it now so I removed comment. :)

Reply:
> Actually, this transformation might be because of this confusing table. ![Image_2024-06-17_03-09-02](https://private-user-images.githubusercontent.com/1018829/340268760-fd259fcb-1a25-4181-81a4-b067d74b76be.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg2Mjg1MjYsIm5iZiI6MTcxODYyODIyNiwicGF0aCI6Ii8xMDE4ODI5LzM0MDI2ODc2MC1mZDI1OWZjYi0xYTI1LTQxODEtODFhNC1iMDY3ZDc0Yjc2YmUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDYxNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA2MTdUMTI0MzQ2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YmY4YjQxMjkzMTI5ZjlhMTZmNWJjYjAzNDkyY2VhOTkzMzVjMjE0NTIwMThmNTNkNjg5OTQxYWQwYmZiNjM4YyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.KOPdJgwFFfGMuEAUebshvkHAW8-KNifv6a64KMCqM9U) So that sounds like you could have found a legitimate bug in our implementation.

The bug in fcmeq was what triggered the test failures once AFP was enabled. I assume you're referring to that? But just changing the order of the ops, fixes it.